ksql-engine/src/test/java/io/confluent/ksql/codegen/SqlToJavaVisitorTest.java

```diff
@@ -75,20 +77,33 @@ public void shouldProcessBasicJavaMath() {
   public void shouldProcessArrayExpressionCorrectly() {
     final String simpleQuery = "SELECT col4[0] FROM test1 WHERE col0 > 100;";
     final Analysis analysis = analyzeQuery(simpleQuery);
+    final String javaExpression = new SqlToJavaVisitor(schema, functionRegistry, ksqlConfig)
+        .process(analysis.getSelectExpressions().get(0));
+
+    assertThat(javaExpression,
+        equalTo("((Double) (ArrayGet.getItem(((java.util.List) TEST1_COL4), (int)(Integer.parseInt(\"0\")), true)))"));
+  }
 
-    final String javaExpression = new SqlToJavaVisitor(schema, functionRegistry)
+
+  @Test
+  public void shouldProcessArrayExpressionCorrectlyForNonLegacyIndexBase() {
```

Comment:
Consider refactoring to make this more readable, e.g. similar to the example I give above in the code runner class.